thirdparty/mbedtls/library/aesni.c

```diff
@@ -41,6 +41,8 @@
 #if MBEDTLS_AESNI_HAVE_CODE == 2
 #if !defined(_WIN32)
 #include <cpuid.h>
+#else
```

Comment:
Can we make sure this gets to mbedtls upstream?

Reply:
I'm not entirely sure whether the issue is in mbedtls or LLVM. The header file `intrin.h` is the Microsoft-only equivalent to `cpuid.h`. However, mbedtls does build on MSVC, which means that header is likely included indirectly. As far as I can tell, there's no path in official MSVC headers from any other `*intrin.h` to `intrin.h`, but there is one from certain concurrency headers (e.g. `concurrent_vector.h`), while in the LLVM headers there are no other headers including `intrin.h` at all.

Another possibility is that MSVC simply accepts `__cpuid` as an implicit definition, and LLVM rejects it on standards grounds. But I think it would at least trigger a warning in this case.

Reply:
Should be reported to upstream mbedtls in case.
Patches also needs to be stored in `thirdparty/mbedtls/patches/` and listed in `thirdparty/README.md` so maintainers can re-apply them when updating the third party code.